karabo_bridge/simulation.py

```diff
@@ -44,21 +44,37 @@
 # _SHAPE = (_PULSES, _MODULES, _MOD_X, _MOD_Y)
 
 
-def gen_combined_detector_data(source, tid_counter):
+def gen_combined_detector_data(source, tid_counter, corrected=False):
     gen = {source: {}}
 
     # metadata
     ts = time()
     sec, frac = str(ts).split('.')
-    gen[source]['metadata'] = {
-        'source': source,
-        'timestamp': ts,
-        'timestamp.tid': tid_counter,
-        'timestamp.sec': sec,
-        'timestamp.frac': frac.ljust(18, '0')  # attosecond resolution
+    gen[source] = {
+        'metadata.source': source,
+        'metadata.timestamp': ts,
+        'metadata.timestamp.tid': tid_counter,
+        'metadata.timestamp.sec': sec,
+        'metadata.timestamp.frac': frac.ljust(18, '0')  # attosecond resolution
     }
 
     # detector random data
+    if corrected:
+        global _PULSES
+        _PULSES = 31
+        global _SHAPE
+        _SHAPE = (_PULSES, _MODULES, _MOD_X, _MOD_Y)
+
+        gain_data = np.zeros(_SHAPE, dtype=np.uint16)
+        passport = [
+            'SPB_DET_AGIPD1M-1/CAL/THRESHOLDING_Q3M2',
+            'SPB_DET_AGIPD1M-1/CAL/OFFSET_CORR_Q3M2',
+            'SPB_DET_AGIPD1M-1/CAL/RELGAIN_CORR_Q3M2'
```

Comment:
This looks like there "where has it been" thing we were talking about. Does this exist now? Is it only for detector images?

Reply:
Well, this is specific to calibration devices, it tells what is applied to the data and will be present when you get corrected data. The more generic "passport" is not implemented yet.